### PR TITLE
[ROOT-10472] Add missing dependencies for coreclingutilstestUnit

### PR DESCRIPTION
**Pull request by oshadura**

### CODE REVIEW

**Comment by chrisburr** [comment]
This hasn't fixed [ROOT-10472](https://sft.its.cern.ch/jira/browse/ROOT-10472) but it did make me notice that `-Dbuiltin_clang=OFF` triggers these two for linking cling itself:

https://github.com/root-project/root/blob/407dd686240427022bcdac17fc49e679dce64030/core/rootcling_stage1/CMakeLists.txt#L21-L31
https://github.com/root-project/root/blob/f7bcad0c573153d94f2a3f3577a70df553a18c77/core/metacling/src/CMakeLists.txt#L68-L78

File: core/clingutils/test/CMakeLists.txt [hunk 1]
```diff
@@ -26,4 +26,4 @@ if(APPLE)
   set(CMAKE_EXE_LINKER_FLAGS "${CMAKE_EXE_LINKER_FLAGS} -w")
 endif()
 
-ROOT_ADD_UNITTEST_DIR(Core RIO ${CLING_LIBRARIES} $<TARGET_OBJECTS:ClingUtils>)
+ROOT_ADD_UNITTEST_DIR(Core RIO ${CLING_LIBRARIES} ${CLING_DEPEND_LIBS} $<TARGET_OBJECTS:ClingUtils>)
```
[the remark this answers is not in the document]
Comment (chrisburr) [reply core/clingutils/test/CMakeLists.txt]: See here: https://sft.its.cern.ch/jira/browse/ROOT-10472

If you look at the gist you'll need to scroll down to the second part that starts with `@@ -1,4 +1,5 @@` as I made a mess of it yesterday.

[the remark this answers is not in the document]
Comment (chrisburr) [reply core/clingutils/test/CMakeLists.txt]: This commit helps but still doesn't completely fix it: https://github.com/root-project/root/commit/6190ea5d17a4ad448dc85183ef7ef4eb082ed6a6

```
$CONDA_PREFIX/lib/libclangCodeGen.a(CGBuiltin.cpp.o):CGBuiltin.cpp:function clang::CodeGen::CodeGenFunction::EmitBuiltinExpr(clang::FunctionDecl const*, unsigned int, clang::CallExpr const*, clang::CodeGen::ReturnValueSlot): error: undefined reference to 'clang::analyze_os_log::computeOSLogBufferLayout(clang::ASTContext&, clang::CallExpr const*, clang::analyze_os_log::OSLogBufferLayout&)'
$CONDA_PREFIX/lib/libclangCodeGen.a(CGBuiltin.cpp.o):CGBuiltin.cpp:function clang::CodeGen::CodeGenFunction::EmitBuiltinExpr(clang::FunctionDecl const*, unsigned int, clang::CallExpr const*, clang::CodeGen::ReturnValueSlot): error: undefined reference to 'clang::analyze_os_log::computeOSLogBufferLayout(clang::ASTContext&, clang::CallExpr const*, clang::analyze_os_log::OSLogBufferLayout&)'
$CONDA_PREFIX/lib/libclangFrontend.a(DiagnosticRenderer.cpp.o):DiagnosticRenderer.cpp:function clang::DiagnosticRenderer::emitDiagnostic(clang::FullSourceLoc, clang::DiagnosticsEngine::Level, llvm::StringRef, llvm::ArrayRef<clang::CharSourceRange>, llvm::ArrayRef<clang::FixItHint>, llvm::PointerUnion<clang::Diagnostic const*, clang::StoredDiagnostic const*>): error: undefined reference to 'clang::edit::Commit::remove(clang::CharSourceRange)'
$CONDA_PREFIX/lib/libclangFrontend.a(DiagnosticRenderer.cpp.o):DiagnosticRenderer.cpp:function clang::DiagnosticRenderer::emitDiagnostic(clang::FullSourceLoc, clang::DiagnosticsEngine::Level, llvm::StringRef, llvm::ArrayRef<clang::CharSourceRange>, llvm::ArrayRef<clang::FixItHint>, llvm::PointerUnion<clang::Diagnostic const*, clang::StoredDiagnostic const*>): error: undefined reference to 'clang::edit::Commit::replace(clang::CharSourceRange, llvm::StringRef)'
$CONDA_PREFIX/lib/libclangFrontend.a(DiagnosticRenderer.cpp.o):DiagnosticRenderer.cpp:function clang::DiagnosticRenderer::emitDiagnostic(clang::FullSourceLoc, clang::DiagnosticsEngine::Level, llvm::StringRef, llvm::ArrayRef<clang::CharSourceRange>, llvm::ArrayRef<clang::FixItHint>, llvm::PointerUnion<clang::Diagnostic const*, clang::StoredDiagnostic const*>): error: undefined reference to 'clang::edit::EditedSource::commit(clang::edit::Commit const&)'
$CONDA_PREFIX/lib/libclangFrontend.a(DiagnosticRenderer.cpp.o):DiagnosticRenderer.cpp:function clang::DiagnosticRenderer::emitDiagnostic(clang::FullSourceLoc, clang::DiagnosticsEngine::Level, llvm::StringRef, llvm::ArrayRef<clang::CharSourceRange>, llvm::ArrayRef<clang::FixItHint>, llvm::PointerUnion<clang::Diagnostic const*, clang::StoredDiagnostic const*>): error: undefined reference to 'clang::edit::EditedSource::applyRewrites(clang::edit::EditsReceiver&, bool)'
$CONDA_PREFIX/lib/libclangFrontend.a(DiagnosticRenderer.cpp.o):DiagnosticRenderer.cpp:function clang::DiagnosticRenderer::emitDiagnostic(clang::FullSourceLoc, clang::DiagnosticsEngine::Level, llvm::StringRef, llvm::ArrayRef<clang::CharSourceRange>, llvm::ArrayRef<clang::FixItHint>, llvm::PointerUnion<clang::Diagnostic const*, clang::StoredDiagnostic const*>): error: undefined reference to 'clang::edit::Commit::insertFromRange(clang::SourceLocation, clang::CharSourceRange, bool, bool)'
$CONDA_PREFIX/lib/libclangFrontend.a(DiagnosticRenderer.cpp.o):DiagnosticRenderer.cpp:function clang::DiagnosticRenderer::emitDiagnostic(clang::FullSourceLoc, clang::DiagnosticsEngine::Level, llvm::StringRef, llvm::ArrayRef<clang::CharSourceRange>,
```
(truncated)

[the remark this answers is not in the document]
Comment (Axel-Naumann) [reply core/clingutils/test/CMakeLists.txt]: Argh sorry, I overlooked "[ROOT-10472]" in the title! Thanks.

[the remark this answers is not in the document]
Comment (Axel-Naumann) [reply core/clingutils/test/CMakeLists.txt]: > This commit helps but still doesn't completely fix it: 6190ea5

Those should be in `-lclangAnalysis` and `-lclangEdit` which are part of `CLING_DEPEND_LIBS`. Could you check whether maybe or order is wrong? Looks like `clangAnalysis` must come after `clangCodeGen`, `clangEdit` after `clangFrontend` (in interpreter/cling/lib/Interpreter/CMakeLists.txt)?

**Comment by oshadura** [comment]
@chrisburr please let us know does it work for you or just open new PR :)

**Comment by oshadura** [comment]
Closing in favor of newer PRs opened by @chrisburr

**Pull request closed without merging**